Complete prework (#10)
Vets-Who-Code/Prework: README.md

<h1 align="center"><img src="/img/vwc.gif" alt="Vets Who Code" width="100px" /> VETS WHO CODE</h1>

---

## [Prework](https://vetswhocode.io)

> Welcome to the official Vets Who Code Prework! Here you'll find the tools, work, and accounts to prepare you for Vets Who Code's prior coursework.  We recommend you complete each section as they build upon one another. The prework should take a total of eight hours to complete.

Reviewer: > The prework should take a total of eight hours to complete.

Should we have hour estimates marked next to each section?

Author: That's not necessary for this. In the coursework that will auto-populate but if you're doing the pre-work it's fine.

> Welcome to the official Vets Who Code Prework! Here you'll find the tools, work, and accounts to prepare you for Vets Who Code's prior coursework.  We recommend you complete each section as they build upon one another. The prework should take a total of eight hours to complete.

Reviewer: ```suggestion
Welcome to the official Vets Who Code Prework! Here you'll find the tools, work, and accounts to prepare you for Vets Who Code's coursework.  We recommend you complete each section as they build upon one another. The prework should take a total of eight hours to complete.
```

 ### Accounts

> Here are the accounts you will need to store your projects, solve problems, prototype, network, write, and be inspired by others' work. You'll also be able to see some fantastic projects from others.

Reviewer: Do students need to have these accounts setup? Are we going to insure they're setup in anyway? At the very bottom there is a call out that the code doesn't HAVE to be on Github. Do we need to update the Prework form with urls for Dev.to, Github, and Linkedin?

#### [LinkedIn For Veterans](https://socialimpact.linkedin.com/programs/veterans/premiumform)

> As a member of the U.S. military community, you can receive one year of LinkedIn Premium through SheerID's collaboration with LinkedIn. This service offers unlimited access to more than 10,000 courses through our LinkedIn Learning platform as a part of the program.

Reviewer: > through SheerID's collaboration with LinkedIn

Why does this matter for our students? Do we have a deal with Linkedin that we're going to advertise their partnerships?

> "our" LinkedIn Learning platform

Can you elaborate on what this means?

Author: SheerId is the tool used to identify their status. It's to let them know what is pulling their data from the government. We have had a partnership with LinkedIn Learning finalized since Jan. Since it's under their program, 'this' refers to LinkedIn Premium. We can change that language to theirs if it's easier for you to understand.

> LinkedIn Learning is also pleased to offer eligible military spouses one year of access to LinkedIn Premium. As part of the program, you'll also gain access to more than 17,000 courses through LinkedIn Learning to help you manage your career and nurture your professional development.

Reviewer: > As part of the program, you'll also gain access

Is this as part of "the program" referring to "as a part of VWC" or as a part of "Linkedin Learning Premium". 

> spouses one year of access to LinkedIn Premium. [...] more than 17,000 courses

Why do spouses get access to 17,000 courses while veterans get access to 10,000 courses. Could we combine military community and their spouses into a single sentence?

Author: The program is about LinkedIn Premium's Social impact program. They also haven't updated their language. It can be updated and merged.

LinkedIn Premium is a subscription-based service that offers users additional features not available on the free LinkedIn platform. These features include seeing who has viewed your profile, sending unlimited InMails to recruiters, and accessing exclusive content and events.

> LinkedIn is a great way to connect with people with similar professional interests and network with potential hiring partners. We also have a LinkedIn Group that we would love for you to join!

Reviewer: > We also have a LinkedIn Group that we would love for you to join!

This is a Call To Action (CTA), there should be a link to the group here.

Author: Copy, will link.

#### [GitHub](https://github.com/)

GitHub is a code hosting platform for version control and collaboration. It lets you and others work together on projects from anywhere.

You can use GitHub to build software alongside millions of other developers. You can also use it to learn how to code, or find help with programming problems.

> GitHub offers an easy way to share your code with friends and classmates, or collaborate on open-source projects.

Reviewer: These 3 sentences seem to be saying the same thing.

Author: I will change the language to "GitHub is a web-based platform that allows you to store and manage your code repositories (collections of code) and collaborate with others. It uses the Git version control system to track changes in your code and provides a user-friendly interface for managing your code and collaborating with others. Developers and organizations widely use GitHub to host and share open-source projects, work on internal software projects, and showcase their work to the broader community."

#### [Dev.to](https://dev.to/)

If you're a developer looking for a career change or an aspiring developer who wants to learn more, Dev.to is the place to start.

Dev.to is an online community where developers from all backgrounds can share their experiences and expertise. It's also the perfect place for anyone interested in learning about software development to connect with like-minded individuals working in the field.

You can find jobs on Dev.to, as well as connect with recruiters and companies seeking employees with your skillset. You can also learn about new technologies, get advice on making the most of your experience and career path, and share tips on success in this industry.

Whether you're looking for a new job or want to meet other developers, Dev.to is an essential part of any aspiring developer's toolkit!
 

> At #VetsWhoCode We Believe that long-form writing is one of the best ways to scale yourself, so you don't have to retain information, showcase how you communicate, and a simple way to advocate for yourself to non-technical employers when you're not writing since everyone on here is in tech so the people will be your audience.

Reviewer: I think that at VWC we see the most success through built networks and a visible online presence. This could take the form of writing, but is not exclusive to writing. I think whether blogs, email lists, Linkedin posts, tik toks, youtube videos, Twitch streaming, meetups, conferences, what matters is being intentional about building your network.

### Command Line

> The command line is a text-based interface for interacting with your computer. It's where you can type in commands to perform tasks like copying files and managing folders and running applications.

Reviewer: Hemingway ain't a fan 
![image](https://user-images.githubusercontent.com/24581531/218546002-bc7aa1bd-52fe-489f-82a8-10092554e5c7.png)

> The command line is a text-based interface for interacting with your computer. It's where you can type in commands to perform tasks like copying files and managing folders and running applications.

Reviewer: ```suggestion
The command line is a text-based interface for interacting with your computer. It's where you can type in commands to perform tasks like copying files, managing folders, and running applications.
```

> It's often called the "command prompt" or "terminal," but it's also known as the "console" or "CLI," which stands for "command line interface." The application name used to access the command line differs from OS to OS: On Windows, it's called cmd (for Command Prompt), while on Mac, it's called Terminal.

Reviewer: For WIndows you could use `Windows Terminal` or `Powershell`. Also, you get setup Git Bash to run *Nix types commands. Also, if someone wants to use Linux I'm assuming they could? Or do they need to buy/setup a Windows/Mac machine?

Following will be the directions to operate in the command line for MacOS and Windows.

### Command Line Operations 

#### Open
You can use the Terminal app to access your computer's command line. The Terminal is one of the most powerful tools on your Mac, and it's how we'll be running our code.

MacOS: To open Terminal, go to Applications, then to Utilities finally to Terminal.

> Windows: To open the command prompt, click Start and then click All Programs. In the programs list, click Accessories. In the accessories list, click Command Prompt.

Reviewer: If we're going to have a Command Line tutorial, I'd assume we'd want 2 drop downs. 1 drop down for the tutorial with Windows and 1 for the tutorial for Mac.

#### Print Working Directory 

> To help you know where you are currently in the Terminal we need to print the wordking directory. A directory would look like a folder on the screen of the GUI( Graphical User Interface). Type this command and hit enter:

Reviewer: > GUI( Graphical User Interface)

Instead 
`Graphical User Interface (GUI)`

> To help you know where you are currently in the Terminal we need to print the wordking directory. A directory would look like a folder on the screen of the GUI( Graphical User Interface). Type this command and hit enter:

Reviewer: ```suggestion
To help you know where you are currently in the Terminal we need to print the working directory. A directory would look like a folder on the screen of the Graphical User Interface (GUI). Type this command and hit enter:
```

MacOS: `pwd`
Windows: `cd`

#### List Directories
To list all the directories and files in your current location type:
MacOS : `ls`
Windows: `dir`

#### Make Directory

We want a place where we can put all our work in. While in the terminal make a new directory called VetsWhoCode by typing mkdir in Windows or MacOS and press enter:

> `mkdir VetsWhoCode`

Reviewer: Where do you want this directory? Students could end up putting it at `/`. I suggest making a `~/code` and `~/code/vetswhocode/` folder. I'd also suggest not using TitleCasing and instead kebab or all lower case. Casing folder names that you need to access via cli gets annoying depending on what computer you're on.

#### Change Working Directory

Now that we have made a new directory lets go there where we will put our work on our machine. te do that by in both windows and MacOS we do this by dyping cd then pressing enter:

`cd VetsWhoCode`

#### Change to previous Directory
If you want to go back to the previous directory for both Windows and MacOS you type this and press enter:

`cd ..`

#### Delete a directory

> Let's say you make a typo and need to delete a directory. Here are the commands to do that.

Reviewer: > make a typo

add, `make a typo in a folder name`

Also, please add a warning that `rm -rf` should not be taken lightly. You have the ability to delete important files that could screw up your computer. If this makes you nervous, it's okay to open the file explorer and right-click delete a folder.

Author: Copy

> MacOS; `rm -r NameOfDirectory`

Reviewer: > MacOS;

instead: `MacOS:`

Also, sometimes you'll need the `-f` and use `rm -rf`

Windows: `rmdir /S NameOfDirectory`

### Code Editor

> An Integrated Development Environment (IDE) is a software application that provides comprehensive tools and features to assist developers in writing, debugging, and deploying software. An IDE typically includes a code editor, a debugger, a compiler or interpreter, and other tools such as a profiler, a version control system, and a build automation tool.

Reviewer: Should this list of features be a bulleted point list?

- code editor
- debugger
- compiler/interpreter
- profiler
- version control system
- build automation tool
- auto-completion
- syntax highlighting

Author: Since it's a natural language, I used commas. I can make it a list if easier to read.

> An IDE can also provide features like auto-completion, syntax highlighting, and code refactoring to make coding faster and more efficient. IDEs are commonly used for developing software applications, web applications, and mobile apps. Our preferred IDE is Visual Studio Code. Download it [here](https://code.visualstudio.com/). You can also install our developer extension pack by going to extensions and typing [VetsWhoCode Extension Pack](https://marketplace.visualstudio.com/items?itemName=VetsWhoCode.vetswhocode-extension-pack).

Reviewer: > You can also install our developer extension pack

Change this to more active voice since it's a CTA

`Install our developer extension pack`,

Author: Will change to " To install our developer extension pack, go to extensions and type [VetsWhoCode Extension Pack](https://marketplace.visualstudio.com/items?itemName=VetsWhoCode.vetswhocode-extension-pack)."

### Browser

> The preferred browser for Vets Who Code is the Google Chrome, due to its popularity and its robust developer tools. To download it please go [here](https://www.google.com/chrome/) and follow the prompts.

Reviewer: > preferred browser for Vets Who Code is the Google Chrome,

remove `the`, it's just `Google Chrome`

> The preferred browser for Vets Who Code is the Google Chrome, due to its popularity and its robust developer tools. To download it please go [here](https://www.google.com/chrome/) and follow the prompts.

Reviewer: ```suggestion
The preferred browser for Vets Who Code is Google Chrome, due to its popularity and its robust developer tools. To download it please go [here](https://www.google.com/chrome/) and follow the prompts.
```

### Using GitHub

>  We use GitHub extensively here. Below you’ll find the steps to use it within the terminal or command prompt.

Reviewer: > We use GitHub extensively here.

What features do we use for Github here? What comes to mind is Version control, access control, ticket management, triage, and collaboration.

Author: We can add them.

> 1.	 Install Git: Git is a version control system used by GitHub. You can download and install Git for Windows from the official Git website.

Reviewer: I thought we preferred the Github CLI over `git` cli. Also, what's our opinion on Github desktop?

Author: Since GitHub is built on it, you still have to install git on your machine. We will teach him about GitHub CLI. I am not a fan of GitHub Desktop however, I won't stop a person from using it. Just don't want to waste man-hours teaching a GUI.

2.	 Configure Git: After installing Git, open the Git Bash application and configure your name and email address using the following commands:
```console
git config --global user.name "Your Name"
git config --global user.email "youremail@example.com"
```

> 3.	Create a repository: A repository is a place where you can store and manage your code. Click on the “New” button and create a new repository.

Reviewer: Nit: Screenshot of what the `New` button looks like?

> 3.	Create a repository: A repository is a place where you can store and manage your code. Click on the “New” button and create a new repository.

Reviewer: ![image](https://user-images.githubusercontent.com/24581531/218554657-1a4750d5-f1a7-4245-96e4-b9205ea8998d.png)
```suggestion
3.	Create a repository: A repository is a place where you can store and manage your code. Click on the “New” button and create a new repository.
![image](https://user-images.githubusercontent.com/24581531/218554688-7d3594a4-eb28-41f2-8683-8426d2783480.png)

```

> 4.	Clone the repository: To start working with the repository, you need to clone it to your local machine. You can do this by clicking on the “Code” button and copying the repository URL. Then, open your terminal or command prompt and type `git clone URL-Goes-Here`.

Reviewer: "Clone" is a new term at this point. I retitle this step to `Download the repository` and explain: `To start working with the repository, you need to download it to your local machine. Github calls this cloning`

Author: We can keep the same language and just explain what cloning is in the beginning.

> 5.	Make changes to the repository: Once you have cloned the repository, you can change the code using your favorite code editor.

Reviewer: > Once you have cloned the repository, you can change the code using your favorite code editor.

Could use more explanation. I suggest

Start coding: Once you have cloned the repository, you can change the code using your favorite code editor. To do this, open up the repository you just downloaded with VS Code (it's just a folder). Make text changes in files, create/remove folders and update their names. You're now coding.

> 5.	Make changes to the repository: Once you have cloned the repository, you can change the code using your favorite code editor.

Reviewer: ```suggestion
5.	Start coding: Once you have cloned the repository, you can change the code using your favorite code editor. To do this, open up the repository you just downloaded with VS Code (it's just a folder). Make text changes in files, create/remove folders and update their names. You're now coding.
```

> 6.	Add changes to the staging area: To add your changes to the staging area, type `git add` followed by the file name or directory you want to add. You can use `git add .` to add all the changes.

Reviewer: I'd update to:

Stage your changes: You need to let `git` know what changes you made. To add your changes to the staging area, type `git add` followed by the file name or directory you want to add. You can use `git add .` to add all the changes. To make check what changes are staged you can run `git status`. In the output the file names in green are the changes that have been staged.
![image](https://user-images.githubusercontent.com/24581531/218550920-6430e103-d36d-472a-b6a2-c8445d72b338.png)

> 6.	Add changes to the staging area: To add your changes to the staging area, type `git add` followed by the file name or directory you want to add. You can use `git add .` to add all the changes.

Reviewer: ```suggestion
6.	Stage your changes: You need to let `git` know what changes you made. To add your changes to the staging area, type `git add` followed by the file name or directory you want to add. You can use `git add .` to add all the changes. To make check what changes are staged you can run `git status`. In the output the file names in green are the changes that have been staged.
![image](https://user-images.githubusercontent.com/24581531/218550920-6430e103-d36d-472a-b6a2-c8445d72b338.png)
```

> 7.	Commit changes: Once you have added your changes to the staging area, you can commit them by typing `git commit -m` followed by a brief description of the changes you made.

Reviewer: Add example syntax

`git commit -m "<commit message>"`

> 7.	Commit changes: Once you have added your changes to the staging area, you can commit them by typing `git commit -m` followed by a brief description of the changes you made.

Reviewer: ```suggestion
7.	Commit changes: Once you have added your changes to the staging area, you can commit them by typing `git commit -m "<commit message>"` followed by a brief description of the changes you made.
```

> 8.	Push changes: To upload your changes to the GitHub repository, type `git push` followed by the name of the branch to which you want to push the changes.

Reviewer: This could be `git push origin main`, I'd suggest `git push` and then look at the output and it should say what command to use, something like `git push --set-upstream origin main`

> 8.	Push changes: To upload your changes to the GitHub repository, type `git push` followed by the name of the branch to which you want to push the changes.

Reviewer: ```suggestion
8.	Push changes: To upload your changes to the GitHub repository, type `git push`. If you're branch isn't setup check the output in the command line it may have a message like `git push --set-upstream origin <branch name>`. Once you've setup your upstream, you can use `git push` for all future uploads to this branch.
```

> 9.	Pull changes: If someone else has changed the repository, you can pull their changes to your local machine by typing `git pull`.

Reviewer: I don't think at this point students should be expecting to work on repos with others.

Author: You never know. Sometimes they have friends who help them. Its happened a few times.

> 9.	Pull changes: If someone else has changed the repository, you can pull their changes to your local machine by typing `git pull`.

Reviewer: ```suggestion
9.	Pull changes: If you've made changes on another computer to update your local machine run `git pull`.
```

10.	Create a pull request: If you want to suggest changes to someone else’s repository, you can create a pull request by clicking on the “Pull request” button on their repository page.

> 11.	Merge changes: The repository owner can merge your changes into their repository by clicking on the “Merge pull request” button.

Reviewer: This guide hasn't mentioned Forking, brancing, or any type of Git Flow at this point. I'd remove these two points.

Author: We are going to teach Git Flow separately. I stated earlier that sometimes the troops have friends or other mentors helping them or reviewing their work. Although they might not use it, it is an essential skill that they should at least know of.

<h2 id="html">HTML & CSS</h2>

> ### HTML && CSS

Reviewer: Do we need to have both of these headers?

> <h2 id="html">HTML & CSS</h2>
> ### HTML && CSS

Reviewer: ```suggestion
### HTML && CSS
```

> HTML (Hypertext Markup Language) is the standard markup language used to create web pages, while CSS (Cascading Style Sheets) is used to style and layout the content. Together, they form the building blocks of most websites.

Reviewer: > HTML (Hypertext Markup Language)

to `Hypertext Markup Language (HTML)`

> is the standard markup language

I'd just say `standard language`

> CSS (Cascading Style Sheets)

to `Cascading Style Sheets (CSS)`

> HTML (Hypertext Markup Language) is the standard markup language used to create web pages, while CSS (Cascading Style Sheets) is used to style and layout the content. Together, they form the building blocks of most websites.

Reviewer: ```suggestion Hypertext Markup Language(HTML) is the standard markup language used to create web pages, while Cascading Style Sheets (CSS) is used to style and layout the content. Together, they form the building blocks of most websites.
```

> HTML (Hypertext Markup Language) is the standard markup language used to create web pages, while CSS (Cascading Style Sheets) is used to style and layout the content. Together, they form the building blocks of most websites.

Reviewer: ```suggestion
Hypertext Markup Language (HTML) is the standard language used to create web pages, while Cascading Style Sheets (CSS) is used to style and layout the content. Together, they form the building blocks of most websites.
```

> 1. Set up your environment: To create an HTML and CSS document, you'll need a text editor and a web browser. You can use any text editor, such as Notepad, Sublime Text, or Atom. You can also use online editors like CodePen or JSFiddle to get started quickly.

Reviewer: Previously we said that you need to be using VS Code however, this has no mention of VS Code.

Author: We can switch over to VS code. Probably copy.ai or chatgpt included those.

2. Create an HTML document: Open your text editor and create a new file. Save it with a .html extension. In this file, you will create the structure of your web page using HTML tags. Here's an example of a basic HTML document:

```
<!DOCTYPE html>
<html>
<head>
	<title>My First Web Page</title>
</head>
<body>
	<h1>Welcome to my website!</h1>
	<p>This is my first web page.</p>
</body>
</html>
```

3. Add content to the HTML document: Inside the body tags, you can add different types of content, such as headings, paragraphs, images, links, and lists. Here are a few examples:

```
<h1>This is a heading</h1>
<p>This is a paragraph.</p>
<img src="image.jpg" alt="An image">
<a href="https://www.example.com">Visit Example</a>
<ul>
	<li>Item 1</li>
	<li>Item 2</li>
	<li>Item 3</li>
</ul>
```

4. Create a CSS file: Create a new file in your text editor and save it with a .css extension. In this file, you will define the styles for your HTML document using CSS rules. Here's an example of a basic CSS rule:
```
h1 {
  color: red;
}
```

This rule sets the color of all h1 elements to red.

> 5 .Link the CSS file to the HTML document: In the head section of your HTML document, add a link to your CSS file using the following code:

Reviewer: ```suggestion
5. Link the CSS file to the HTML document: In the head section of your HTML document, add a link to your CSS file using the following code:
```
